docs/recipes/Zustand.md

```diff
@@ -0,0 +1,688 @@
+import { EpisodeStore, createEpisodeSlice, episodeStoreSelector } from "./EpisodeStore";
+
+export interface RootStore extends AuthenticationStore, EpisodeStore {}
+
+export const useStore = create<RootStore>()((...a) => ({
+  ...createAuthenticationSlice(...a),
+  ...createEpisodeSlice(...a),
+  // add your state slices here
+}));
+
+// optional: custom hooks can be used to pick pieces from state
+// useShallow is used to help prevent unnecessary rerenders
+export const useAuthenticationStore = () => useStore(useShallow(authenticationStoreSelector));
+export const useEpisodeStore = () => useStore(useShallow(episodeStoreSelector));
+```
+
+- We're combining AuthenticationStore and EpisodeStore into one Zustand store for simplicity. Again, Zustand is very non-opinionated so you can modify this structure if desired.
+- `useAuthenticationStore` and `useEpisodeStore` are exported as custom hooks to make it easier to select common pieces of state. Feel free to create additional custom hooks for reusable lookup patterns and prevent unnecessary re-renders [Read more about this](https://github.com/pmndrs/zustand?tab=readme-ov-file#selecting-multiple-state-slices).
+
+Create `store/index.ts` file to export our hooks and selectors for easy use across our app:
+
+**`app/store/index.ts`**
+
+```ts
+export * from "./RootStore";
+export * from "./AuthenticationStore";
+export * from "./EpisodeStore";
+export * from "./Episode";
+```
+
+## Use Zustand in Components
+
+Zustand's hooks-based API makes it easy to pull data into components.
+
+In the Ignite Demo App, we'll update the following components to use our exported Zustand hooks and selectors:
+
+**`app/navigators/AppNavigator.tsx`**
+
+```tsx
+import { useStore, isAuthenticatedSelector } from "../store";
+
+const AppStack = () => {
+
+  // use a selector to pick only that value
+  const isAuthenticated = useStore(isAuthenticatedSelector)
+
+  return (
+    <Stack.Navigator
+    ...
+```
+
+**`app/screens/LoginScreen.tsx`**
+
+```tsx
+// pick several values & actions from the AuthenticationStore
+const { authEmail, setAuthEmail, setAuthToken } = useAuthenticationStore();
+// we can also use multiple hooks
+const validationError = useStore(validationErrorSelector);
+```
+
+**`app/screens/DemoPodcastListScreen.tsx`**
+
+Several changes are needed here. We'll use `useEpisodeStore` to select all the data and actions we need from EpisodeStore:
+
+```ts
+import {
+  useEpisodeStore,
+  Episode,
+  getDatePublished,
+  getDuration,
+  getParsedTitleAndSubtitle,
+} from "app/store"
+
+...
+
+const episodeStore = useEpisodeStore();
+```
+
+We also need to update how we get derived values from `Episode` now that we're working with a plain object in Zustand without custom getters. Instead of `episode.duration` we can use our util function `getDatePublished`. Add these lines to the render function of `EpisodeCard`, and replace a few spots those values are used.
```

Comment:
In addition to this, there is also a change to the `onValueChange` in the `ListHeaderComponent` for toggling the favorites only part of the list since `episodeStore.setProp` no longer exists

```
onValueChange={() => episodeStore.setFavoritesOnly(!episodeStore.favoritesOnly)}
```

Reply:
good catch!